src/sentry/integrations/github/blame.py

```diff
@@ -138,77 +151,107 @@ def extract_commits_from_blame_response(
                         },
                     )
                     continue
-                matching_file, blame_info = _get_matching_file_and_blame(
-                    files=files,
-                    blame_ranges=blame.get("ranges", []),
-                    path=file_path,
-                    repo_name=full_repo_name,
-                    ref=ref_name,
-                )
-                if not blame_info:
-                    logger.error(
-                        "get_blame_for_files.extract_commits_from_blame.missing_line_blame",
-                        extra={
-                            **extra,
-                            "file_lineno": matching_file.lineno,
-                            "file_path": matching_file.path,
-                            "branch_name": matching_file.ref,
-                            "repo_name": full_repo_name,
-                        },
+                matching_files = [
+                    f
+                    for f in files
+                    if f.path == file_path and f.repo.name == full_repo_name and f.ref == ref_name
+                ]
+                for file in matching_files:
```

Comment:
Made a small unrelated change here. I realized that multiple frames can map to the same file if they have  different line numbers, so instead of getting the first file we need to map through each matching file. Made a small modification to a test to ensure that this works.